docs_src/design/adr/0006-Metrics-Collection.md

```diff
@@ -0,0 +1,397 @@
+
+### Requirements
+
+- Services will push specified metrics collected for that service to a specified (by configuration) message endpoint (as supported by the EdgeX  message bus implementation; currently either Redis Pub/Sub or MQTT implementations are supported)
+    - Each service will have configuration that specifies a message endpoint for the service metrics.  The metrics message topic communications may be secured or unsecured (just as application services provide the means to export to secured or unsecured message pipes today).
+    - The configuration will be placed in the `Writable` area.  When a user wishes to change the configuration dynamically (such as turning on/off a metric), then Consul's UI can be used to change it.
+- Services will have configuration which indicates what metrics are available from the service.
+- Services will have configuration which allows EdgeX system managers to select which metrics are `on` or `off` - in other words providing configuration that determines what metrics are collected and reported by default.
+    - When a metric is turned `off` (the default setting) the service does not report the metric.  When a metric is turned `on` the service collects and sends the metric to the designated message topic.
+    - Metrics collection must be pushed to the designated message topic on some appointed schedule.  The schedule would be designated by configuration and done in a way similar to auto events in device services.
+    - For the initial implementation, there will be just one scheduled time when all metrics will be collected and pushed to the designated message topic.  In the future, there may be a desire to set up a separate schedule for each metric, but this was deemed too complex for the initial implementation.
+
+!!! Info
+    Initially, it was proposed that metrics be associated with a "level" and allow metrics to be turned on or off by level (like levels associated to log messages in logging).  The level of metrics data seems arbitrary at this time and considered too complex for initial implementation.  This may be reconsidered in a future release and based on new requirements/use cases.
+
+    It was also proposed to categorize or label metrics - essentially allowing grouping of various metrics.  This would allow groups of metrics to be turned on or off, and allow metrics to be organized per the group when reporting.  At this time, this feature is also considered beyond the scope of the initial implementation and to be reconsidered in a future release based on requirements/use case needs.
+
+    It was also proposed that each service offer a REST API to provide metrics collection information (such as which metrics were being collected) and the ability to turn the collection on or off dynamically.  This is deemed out of scope for the first implementation and may be brought back if there are use case requirements / demand for it. 

```

Comment:
Formatting sections with this block.  Please view in preview to see.

Reply:
This was as intended Bryon.  All that is under this Info was proposals we wanted to capture outside the rest of the discussion.  I am open to alternative formatting if it helps.